File: s1/s1/src/test/java/com/nodal/s1/consumer/RabbitMQConsumerTest.java
package com.nodal.s1.consumer;

import org.junit.jupiter.api.Test;
import org.springframework.amqp.rabbit.core.RabbitTemplate;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.boot.test.mock.mockito.MockBean;

@SpringBootTest
public class RabbitMQConsumerTest {
    @MockBean
    private RabbitTemplate rabbitTemplate;

    @Autowired
    private RabbitMQConsumer consumer;

    @Test
    public void testConsumeResponse(){
        consumer.consumeResponse("Pong");
    }

    @Test
    public void testConsumeSend(){
        consumer.consumeSend("Ping");
    }
}

File: s2/s2/src/test/kotlin/com/nodal/s2/consumer/RabbitMQConsumerTest.kt
package com.nodal.s2.consumer

import org.junit.jupiter.api.Test
import org.springframework.amqp.rabbit.core.RabbitTemplate
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.boot.test.mock.mockito.MockBean

@SpringBootTest
class RabbitMQConsumerTest {
    @MockBean
    private val rabbitTemplate: RabbitTemplate? = null

    @Autowired
    private lateinit var consumer: RabbitMQConsumer

    @Test
    fun testConsumeResponse() {
        consumer.consumeResponse("Pong")
    }

    @Test
    fun testConsumeSend() {
        consumer.consume("Ping")
    }
}